app/components/primer/alpha/action-menu-element.ts

```diff
@@ -0,0 +1,382 @@
+import type {AnchorAlignment, AnchorSide} from '@primer/behaviors'
+import {getAnchoredPosition} from '@primer/behaviors'
+// WORK IN PROGRESS - DO NOT USE
+
+// Necessary to turn this file into an external module, which enables the Window
+// interface modifications below.
+export {}
+
+class ActionMenuElement extends HTMLElement {
+  #abortController: AbortController
+  // eslint-disable-next-line no-invalid-this
+  #actionMenuEl: HTMLElement = this
+  // eslint-disable-next-line no-invalid-this
+  #buttonEl = this.querySelector<HTMLButtonElement>('button')
+  // eslint-disable-next-line no-invalid-this
+  #menuEl = this.querySelector<HTMLUListElement>('[role="menu"]')
+  #menuItemEls: HTMLElement[] = []
+  #firstMenuItem: HTMLElement
+  #lastMenuItem: HTMLElement
+  #firstCharactersOfItems: string[] = []
+  // eslint-disable-next-line no-undef
+  #allMenuItemEls: NodeListOf<HTMLElement> | null =
+    // eslint-disable-next-line no-invalid-this
+    this.#menuEl && this.#menuEl.querySelectorAll('[role="menuitem"],[role="menuitemcheckbox"],[role="menuitemradio"]')
+
+  get anchorAlign(): AnchorAlignment {
+    return (this.getAttribute('data-anchor-align') || 'start') as AnchorAlignment
+  }
+
+  set anchorAlign(value: AnchorAlignment) {
+    this.setAttribute('data-anchor-align', value)
+  }
+
+  get anchorSide(): AnchorSide {
+    return (this.getAttribute('data-anchor-side') || 'outside-bottom') as AnchorSide
+  }
+
+  set anchorSide(value: AnchorSide) {
+    this.setAttribute('data-anchor-align', value)
+  }
+
+  connectedCallback() {
+    if (!this.#buttonEl) return
+
+    // THIS IS TEMPORARY AND SHOULD BE REPLACED BY PRIMER CSS CLASS
+    const style = document.createElement('style')
+    style.innerHTML = `
+        action-menu {
+          position: relative;
+        }
+        action-menu ul {
+          position: absolute;
+          width: 160px;
+          background-color: var(--color-canvas-overlay);
+          z-index: 1000000;
+          border: $border-width $border-style var(--color-border-default);
+          border-radius: $border-radius;
+          box-shadow: var(--color-shadow-large);
+        }
+    `
+    document.querySelector('body')?.appendChild(style)
```

Comment:
Just temporarily keeping this here since `Overlay` isn't available yet in the previews. 
Can you confirm that all of this can be replaced by `Overlay`? @langermank

Reply:
Yes, exactly! Overlay will replace any inline style